code/modules/mob/living/simple_animal/hostile/gym_rats.dm

```diff
@@ -0,0 +1,703 @@
+	stop_automated_movement_when_pulled = TRUE
+
+	min_oxy = 8 //Require atleast 8kPA oxygen
+	minbodytemp = 223		//Below -50 Degrees Celcius
+	maxbodytemp = 323	//Above 50 Degrees Celcius
+
+	attack_sound = 'sound/weapons/punch1.ogg'
+
+	var/health_cap = 75 // Eating protein can pack on a whopping 150% increase in max health. GAINZ
+	var/icon_eat = "gymrat-eat"
+	var/obj/my_wheel
+	var/list/gym_equipments = list(/obj/structure/stacklifter, /obj/structure/punching_bag, /obj/structure/weightlifter, /obj/machinery/power/treadmill)
+
+	var/static/list/edibles = list(/obj/item/weapon/reagent_containers/food/snacks/cheesewedge, /obj/item/weapon/reagent_containers/food/snacks/meat, /obj/item/weapon/reagent_containers/food/snacks/sliceable/cheesewheel) // Gym rats are pickier than normal mice. Cheese and raw meat only
+
+	var/all_fours = 1
+
+/mob/living/simple_animal/hostile/retaliate/gym_rat/update_icon()
+	if(all_fours == 1)
+		icon_state = "gymrat"
+		icon_eat = "gymrat-eat"
+		attacktext = "headbutts"
+	else
+		icon_state = "gymrat_stand"
+		icon_eat = null
+		attacktext = "punches"
+
+/mob/living/simple_animal/hostile/retaliate/gym_rat/UnarmedAttack(var/atom/A)
+	if(is_type_in_list(A, edibles)) // If we click on something edible, it's time to chow down!
+		delayNextAttack(10)
+		chowdown(A)
+	if(is_type_in_list(A, gym_equipments)) // If we click on gym equipment, it's time to work out!
+		A.attack_hand(src, 0, A.Adjacent(src))
+	else return ..()
+
+/mob/living/simple_animal/hostile/retaliate/gym_rat/proc/chowdown(var/atom/eat_this)
+	if(istype(eat_this,/obj/item/weapon/reagent_containers/food/snacks/cheesewedge)) //Mmm, cheese wedge. Gives back a small amount of health upon consumption
+		health+=5
+		visible_message("\The [name] gobbles up \the [eat_this].", "<span class='notice'>You gobble up the [eat_this].</span>")
+		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
+		flick(icon_eat, src)
+		qdel(eat_this)
+	if(istype(eat_this,/obj/item/weapon/reagent_containers/food/snacks/meat)) //Protein! Gives back a smaller amount of health, but also packs on some extra max hp
+		health+=3
+		maxHealth+=3
+		visible_message("\The [name] gobbles up \the [eat_this].", "<span class='notice'>You gobble up the [eat_this].</span>")
+		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
+		flick(icon_eat, src)
+		qdel(eat_this)
+	if(istype(eat_this,/obj/item/weapon/reagent_containers/food/snacks/sliceable/cheesewheel)) //A cheese wheel feast! Gives back a lot more health than just a slice
+		health+=25
+		visible_message("\The [name] gobbles up \the [eat_this].", "<span class='notice'>You gobble up the [eat_this].</span>")
+		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
+		flick(icon_eat, src)
+		qdel(eat_this)
+
+/mob/living/simple_animal/hostile/retaliate/gym_rat/Life() // Copied from hammy wheel running code
+	if(timestopped)
+		return 0
+	. = ..()
+	if(.)
+		if(enemies.len && prob(5))
+			Calm()
+	if(maxHealth < 60)
```

Comment:
updating these in life has... weird consequences, it means for example that an admin could never bus a gym rat that has DOOR_SMART flags because they will get updated next tick

Reply:
Not really sure of a better way it could be done. Their max health does need to get checked occasionally, in case they get stronger and gain new abilities, or get weaker after getting sprayed with soymilk.

Reply:
have a proc like `adjust_health(var/amount)` that you can call any time the rat is supposed to go up or down in health -- this changes their health and maxhealth and then updates other stats to go along with it. This is more maintainable, too.

Reply:
I might need to sleep on this one. If you have any examples that I could look at, please share them. I'm having a hard time figuring this one out.

Reply:
This should be done!